test/response-schema.spec.ts

```diff
@@ -47,7 +46,6 @@ describe('response schema', () => {
         return reply.code(500).send({
           error: 'Internal Server Error',
           message: "Response doesn't match the schema",
-          details: (err as unknown as ResponseValidationError).details,
```

Comment:
why can't we return details here and assert that validation error exposes enough details?

Reply:
There is no details on ResponseValidationError.

We would now have to access it through error.validation

https://fastify.dev/docs/latest/Reference/Validation-and-Serialization/#error-handling
```
fastify.setErrorHandler(function (error, request, reply) {
  if (error.validation) {
     reply.status(422).send(new Error('validation failed'))
  }
})
```

Reply:
that's fine, can we use that field to preserve the initial test logic?

Reply:
Done, I also renamed ResponseValidationError to ResponseSerializationError since that is what it is.

Reply:
Validation errors can accessed through error.validation as described in the docs.

https://fastify.dev/docs/latest/Reference/Validation-and-Serialization/#error-handling
```
fastify.setErrorHandler(function (error, request, reply) {
  if (error.validation) {
     reply.status(422).send(new Error('validation failed'))
  }
})
```

Reply:
fastify error model really isn't very flexible :-/
trying to augment errors we are getting there currently